analysis/complex.lean

```diff
@@ -201,24 +190,18 @@ rw [abs_mul_abs_self],
   simp,
 end
 
-lemma add_comm : ∀ (a b : ℂ), a + b = b + a := by crunch
-
--- I don't think I ever use these actually.
-local attribute [simp] of_real_zero of_real_one of_real_neg of_real_add
-local attribute [simp] of_real_sub of_real_mul of_real_inv
-
 instance : discrete_field complex :=
 { discrete_field .
-  zero         := 0,
-  add          := (+),
+  add              := (+),
+  zero             := 0,
   neg          := complex.neg,
   zero_add     := by crunch,
   add_zero     := by crunch,
   add_comm     := by crunch,
   add_assoc    := by crunch,
   add_left_neg := by crunch,
```

Comment:
This is a fairly common pattern. Rather than reproving the new properties the second time around, just omit them entirely, and fill them in with `{ discrete_field . <new stuff> , ..complex.add_group }`, where the dot dot notation says "copy the fields from here". See `rat.discrete_linear_ordered_field` for an example.

Reply:
I tried this but then got into a real mess with type class inference. Will try again tomorrow.